### ISSUE TITLE
My rear windows show X as if they are down but they arent

### ISSUE THREAD

**Issue by ryanhutzel1** [tonesto7/fordpass-scriptable#31]
Not sure if everyone is having that issue but I can’t get the check marks to show up on my 2 rear windows. They have an x showing they are down but they aren’t. 2020 F250

**Comment by tonesto7** [comment]
I really wish I had a way of filtering out unsupported doors.  
Do me a favor and tap on the widget to bring up the Widget menu
Tap on Widget Settings > Debug Menu > View Vehicle Data.  (There will be a lot of data to scroll through)
Please share with me the doorStatus section of the large text object.

**Comment by ryanhutzel1** [comment]
Is the is what your looking for?

doorStatus": {
                "rightRearDoor": {
                    "value": "Closed",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "leftRearDoor": {
                    "value": "Closed",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "driverDoor": {
                    "value": "Ajar",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "passengerDoor": {
                    "value": "Closed",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "hoodDoor": {
                    "value": "Closed",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "tailgateDoor": {
                    "value": "Closed",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "innerTailgateDoor": {
                    "value": "Closed",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"

Ryan Hutzel

**Comment by tonesto7** [comment]
Crap!
I meant the windowStatus sorry.

**Comment by ryanhutzel1** [comment]
statusWindows": {
        "leftFront": false,
        "rightFront": false,
        "leftRear": true,
        "rightRear": true

Ryan Hutzel

**Comment by ryanhutzel1** [comment]
All my windows are up right now. 



Ryan Hutzel

**Comment by tonesto7** [comment]
Sorry i need more info.
can you share the output from windowPosition.  
Maybe there is some weird value their thats throwing off the logic

**Comment by ryanhutzel1** [comment]
<html><head><meta name="viewport" content="width=device-width; initial-scale=1.0; maximum-scale=1.0;">
        
            <title>Vehicle Data Page</title>
        </head>
        
        <body>
            <h3 style="color:black;">Vehicle Data</h3>
            <pre id="otaCode_el" style="color:black; font-size: 10px;">{
    "info": {
        "vehicle": {
            "vin": "1FT8W2BT9LED92293",
            "nickName": "",
            "vehicleType": "2020 F-250",
            "color": "STAR WHITE 3C/PRISTINE WHITE 3",
            "modelName": "F-250",
            "modelCode": "VLFE",
            "modelYear": "2020",
            "tcuEnabled": 1,
            "make": "F",
            "cylinders": null,
            "drivetrain": null,
            "engineDisp": "ENTR",
            "fuelType": "D",
            "series": "Crew Cab",
            "productVariant": null,
            "averageMiles": null,
            "estimatedMileage": "0",
            "mileage": "11443",
            "mileageDate": "2021-02-27T11:35:49Z",
            "mileageSource": null,
            "drivingConditionId": null,
            "configurationId": null,
            "primaryIndicator": "",
            "licenseplate": "",
            "purchaseDate": null,
            "registrationDate": "2020-08-09T21:13:12.180Z",
            "ownerCycle": null,
            "ownerindicator": "Y",
            "brandCode": "Ford",
            "vehicleImageId": null,
            "headUnitType": null,
            "steeringWheelType": null,
            "lifeStyleXML": "",
            "syncVehicleIndicator": "",
            "vhrReadyDate": "2021-02-16T13:18:07Z",
            "vhrNotificationDate": "01/01/0001 00:00:00",
            "vhrUrgentNotificationStatus": null,
            "vhrStatus": null,
            "vhrNotificationStatus": null,
            "ngSdnManaged": 1,
            "transmission": "10 Speed Auto Trans 10R140 (RW",
            "bodyStyle": null,
            "preferredDealer": "F24402",
            "assignedDealer": null,
            "sellingDealer": null,
            "vhrReadyIndicator": null,
            "vehicleAuthorizationIndicator": 1,
            "hasAuthorizedUser": 1,
            "latestMileage": "27253",
            "vehicleRole": "OWNER",
            "warrantyStartDate": "2020-08-09T04:00:00Z",
            "versionDescription": "250 SERIES",
            "vehicleUpdateDate": null
        },
        "status": 200,
        "version": "1.0.0"
    },
    "capabilities": [
        "ADBLUE",
        "CEN_LAST_MILE",
        "CEN_OFFBOARD_SEARCH",
        "CEN_SEND_TO_CAR",
        "DISTANCE_TO_EMPTY_TCU",
        "DOOR_LOCK_UNLOCK",
        "EXTEND",
        "FUEL_TCU",
        "ODOMETER_TCU",
        "OIL_LIFE",
        "OIL_LIFE_TCU",
        "REMOTE_START",
        "SCHEDULED_START",
        "TPMS_TCU",
        "VEHICLE_HEALTH_REPORTING",
        "VEHICLE_LOCATOR",
        "VHA2.0_TCU",
        "WIFI_DATA_USAGE",
        "WIFI_HOTSPOT",
        "WIFI_SSID_PASSWORD"
    ],
    "rawStatus": {
        "vehiclestatus": {
            "vin": "1FT8W2BT9LED92293",
            "lockStatus": {
                "value": "UNLOCKED",
                "status": "CURRENT",
                "timestamp": "12-20-2021 13:53:22"
            },
            "alarm": {
                "value": "NOTSET",
                "status": "CURRENT",
                "timestamp": "12-20-2021 13:53:22"
            },
            "PrmtAlarmEvent": {
                "value": "Null",
                "status": "CURRENT",
                "timestamp": "12-20-2021 13:53:22"
            },
            "odometer": {
                "value": 51683,
                "status": "CURRENT",
                "timestamp": "12-20-2021 13:53:22"
            },
            "fuel": {
                "fuelLevel": 35.434896,
                "distanceToEmpty": 342.1,
                "status": "CURRENT",
                "timestamp": "12-20-2021 13:53:22"
            },
            "gps": {
                "latitude": "27.4292317",
                "longitude": "-82.2743600",
                "gpsState": "UNSHIFTED",
                "status": "CURRENT",
                "timestamp": "12-20-2021 13:53:22"
            },
            "remoteStart": {
                "remoteStartDuration": 0,
                "remoteStartTime": 0,
                "status": "CURRENT",
                "timestamp": "12-20-2021 13:53:22"
            },
            "remoteStartStatus": {
                "value": 0,
                "status": "CURRENT",
                "timestamp": "12-20-2021 13:53:22"
            },
            "battery": {
                "batteryHealth": {
                    "value": "STATUS_GOOD",
                    "timestamp": "12-14-2021 14:49:54"
                },
                "batteryStatusActual": {
                    "value": 13,
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                }
            },
            "oil": {
                "oilLife": "STATUS_GOOD",
                "oilLifeActual": 37,
                "status": "CURRENT",
                "timestamp": "12-20-2021 13:53:22"
            },
            "tirePressure": {
                "value": "STATUS_GOOD",
                "timestamp": "09-02-2021 10:47:17"
            },
            "authorization": "AUTHORIZED",
            "TPMS": {
                "tirePressureByLocation": {
                    "value": 1,
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "tirePressureSystemStatus": {
                    "value": "Sensr_Fault_Composite_Stat",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "dualRearWheel": {
                    "value": 0,
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "leftFrontTireStatus": {
                    "value": "Normal",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "leftFrontTirePressure": {
                    "value": "398",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "rightFrontTireStatus": {
                    "value": "Fault",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "rightFrontTirePressure": {
                    "value": "395",
                    "status": "LAST_KNOWN",
                    "timestamp": "12-20-2021 13:23:09"
                },
                "outerLeftRearTireStatus": {
                    "value": "Normal",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "outerLeftRearTirePressure": {
                    "value": "520",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "outerRightRearTireStatus": {
                    "value": "Normal",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "outerRightRearTirePressure": {
                    "value": "517",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "innerLeftRearTireStatus": {
                    "value": "Not_Supported",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "innerLeftRearTirePressure": null,
                "innerRightRearTireStatus": {
                    "value": "Not_Supported",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "innerRightRearTirePressure": null,
                "recommendedFrontTirePressure": {
                    "value": 55,
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "recommendedRearTirePressure": {
                    "value": 50,
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                }
            },
            "firmwareUpgInProgress": {
                "value": false,
                "timestamp": "10-23-2021 14:35:44"
            },
            "deepSleepInProgress": {
                "value": false,
                "timestamp": "12-14-2021 15:01:16"
            },
            "ccsSettings": {
                "timestamp": "08-18-2020 23:09:35",
                "location": 1,
                "vehicleConnectivity": 1,
                "vehicleData": 1,
                "drivingCharacteristics": -1,
                "contacts": -1
            },
            "lastRefresh": "12-20-2021 13:53:09",
            "lastModifiedDate": "12-20-2021 13:53:23",
            "serverTime": "12-20-2021 16:19:21",
            "batteryFillLevel": null,
            "elVehDTE": null,
            "hybridModeStatus": null,
            "chargingStatus": null,
            "plugStatus": null,
            "chargeStartTime": null,
            "chargeEndTime": null,
            "preCondStatusDsply": null,
            "chargerPowertype": null,
            "batteryPerfStatus": null,
            "outandAbout": {
                "value": "PwPckOffTqNotAvailable",
                "status": "CURRENT",
                "timestamp": "12-20-2021 13:53:22"
            },
            "batteryChargeStatus": null,
            "dcFastChargeData": null,
            "windowPosition": {
                "driverWindowPosition": {
                    "value": "Fully_Closed",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "passWindowPosition": {
                    "value": "Fully_Closed",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "rearDriverWindowPos": {
                    "value": "Undefined",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "rearPassWindowPos": {
                    "value": "Undefined",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                }
            },
            "doorStatus": {
                "rightRearDoor": {
                    "value": "Closed",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "leftRearDoor": {
                    "value": "Closed",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "driverDoor": {
                    "value": "Ajar",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "passengerDoor": {
                    "value": "Closed",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "hoodDoor": {
                    "value": "Closed",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "tailgateDoor": {
                    "value": "Closed",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "innerTailgateDoor": {
                    "value": "Closed",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                }
            },
            "ignitionStatus": {
                "value": "Off",
                "status": "CURRENT",
                "timestamp": "12-20-2021 13:53:22"
            },
            "batteryTracLowChargeThreshold": null,
            "battTracLoSocDDsply": null,
            "dieselSystemStatus": {
                "exhaustFluidLevel": {
                    "value": "27.600000",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "filterSoot": {
                    "value": "95",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "ureaRange": {
                    "value": "1100",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "metricType": {
                    "value": "Inactive",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                },
                "filterRegenerationStatus": {
                    "value": "DPF_Normal_Operation",
                    "status": "CURRENT",
                    "timestamp": "12-20-2021 13:53:22"
                }
            }
        },
        "version": "4.0.0",
        "status": 200
    },
    "fetchTime": 1640017162196,
    "evVehicle": null,
    "odometer": 51683,
    "oilLife": 37,
    "lockStatus": "UNLOCKED",
    "ignitionStatus": "Off",
    "zoneLightingSupported": true,
    "zoneLightsStatus": "Off",
    "remoteStartStatus": {
        "running": false,
        "duration": 0
    },
    "alarmStatus": "Off",
    "batteryStatus": "STATUS_GOOD",
    "batteryLevel": 13,
    "firmwareUpdating": false,
    "distanceToEmpty": 342.1,
    "fuelLevel": 35,
    "position": "26110 61st Ave E, Myakka City",
    "latitude": 27.4292317,
    "longitude": -82.27436,
    "statusWindows": {
        "leftFront": false,
        "rightFront": false,
        "leftRear": true,
        "rightRear": true
    },
    "statusDoors": {
        "leftFront": true,
        "rightFront": false,
        "leftRear": false,
        "rightRear": false,
        "hood": false,
        "tailgate": false,
        "innerTailgate": false
    },
    "tirePressure": {
        "leftFront": "57.7",
        "rightFront": "57.3",
        "leftRear": "75.4",
        "rightRear": "75.0"
    }
}</pre>
            <script>
            </script>
        
          
    </body></html>

Ryan Hutzel

**Comment by ryanhutzel1** [comment]
Sorry. Also wanted to mention my LF door is showing as open when it’s closed. Thought you may want to know that also. 

PS. Thanks for writing such an awesome app. I have the widget on my Home Screen at all times. It helps me to remember to lock my truck at night. Great job!!

Ryan Hutzel

**Comment by tonesto7** [comment]
Thanks for the info yours returns different window data than most vehicles. 
I have it fixed in this file: https://gist.githubusercontent.com/tonesto7/942bda23e3228f16f324bf4e68fec6d2/raw/5802c44a6f7054f3659f996bbc7c284cc7db812a/Fordpass%2520Widget.js

Ford does think your driver's door is open.  
Maybe try opening and closing the door in case the state is stuck

**Comment by ryanhutzel1** [comment]
One last question. Is there an easy way to install the new program? Every time there’s an update I have to look at the wrong code and on an iPhone it does not allow you to select all. I have to select the text and drag the cursor for about five minutes to get to the bottom of the page before I can hit copy to copy the source. Just wondered if you knew of an easier way. 

Ryan Hutzel

**Comment by tonesto7** [comment]
I'm working on it... There is a scriptable script called ScriptDude https://scriptdu.de but it won't support updating multiple instances.  So I'm going to have to build my own version of it.

**Comment by ryanhutzel1** [comment]
Awesome. Looks like the new script worked. See attached image. 



Ryan Hutzel

**Comment by ryanhutzel1** [comment]
Hello,
You helped me with a widget issue I had back in December. I’ve been using the widget since then and up until recently have had no issues. When I recently saw there was an update for version 3.14 I attempted to update. What happened was it created a second instance of the widget in the scriptable app. When I clicked the new instance it did nothing. I removed the widget, restarted my phone and attempted to reinstall the widget from the website:

https://tonesto7.github.io/fordpass-scriptable/


When I do this it installs and asks me for my vin and other fordpass info. It then won’t run. Sometimes it will but it will ask me for the vin and info every time I open it. Also the widget doesn’t work on my Home Screen. It only displays the scriptable image and not the fordpass widget. 

Any help would be appreciated. 

Thanks, 
Ryan Hutzel
